src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -575,6 +590,24 @@ protected void retrieve(@CheckForNull SCMSourceCriteria criteria, @NonNull SCMHe
                     @Override
                     protected Iterable<BitbucketBranch> create() {
                         try {
+                            if (event != null) {
```

Comment:
I think I found the culprit. Basically no branch is added if an event is detected but its not of any of the given types.